### PR TITLE
Clear the timer when in error cases and don't update the on download call back

### PR DESCRIPTION
**Pull request by rakeshvanga**

### CODE REVIEW

**Review by lolodi — APPROVED** [review]

File: src/FileDownloader.ts [hunk 1]
```diff
@@ -58,20 +58,20 @@ export default class FileDownloader implements IFileDownloader {
...
+        let progress = 0;
+        const progressTimerId = setInterval(() => {
+            if (progress <= 100) {
+                // TODO: the whole timer should be under this if.
+                if(onDownloadProgressChange != null){
+                    onDownloadProgressChange(progress++, 100);
                 }
-            }, 1500);
+            }
+            else{
```
Comment (daniv-msft) [inline src/FileDownloader.ts:69]: Nit: could you please add a space before the bracket? Same comment above/below for other occurrences. I don't believe that the linter is ran on this repo by default, so feel free to run it manually.

Comment (daniv-msft) [reply src/FileDownloader.ts:69]: Actually, I saw you're not the one who introduced it. Feel free to keep as is.

File: src/FileDownloader.ts [hunk 2]
```diff
@@ -58,20 +58,20 @@ export default class FileDownloader implements IFileDownloader {
         const timeoutInMs = settings?.timeoutInMs ?? DefaultTimeoutInMs;
         const retries = settings?.retries ?? DefaultRetries;
         const retryDelayInMs = settings?.retryDelayInMs ?? DefaultRetryDelayInMs;
-        try{
```
Comment (daniv-msft) [inline src/FileDownloader.ts:61]: We removed the `try catch` from the first part of the code, but wouldn't it make sense to keep catching potential errors from it and log them as errors? Typically, it seems that `onDownloadProgressChange` could easily throw as this is provided by the consumer of the API.

**Review by daniv-msft — APPROVED** [review]
LGTM minus minor comments. Thanks!

**Review by daniv-msft — APPROVED** [review]

**Pull request merged**